megabus/src/main/java/com/bazaarvoice/megabus/refproducer/MegabusRefProducer.java

```diff
@@ -164,6 +174,8 @@ private boolean peekAndAckEvents() {
     }
 
     private void trackAverageEventDuration(long durationInNs, int numEvents) {
+        _eventCounter.inc(numEvents);
```

Comment:
to reduce method invocations we can increment counter after condition
```suggestion
        if (numEvents == 0) {
            return;
        }
        _eventCounter.inc(numEvents);
```

Reply:
Ah, good point. I just assumed Metrics would skip zeroes, but it looks like it won't. Thanks.